src/catkin_pkg/condition.py

```diff
@@ -58,6 +58,20 @@ def _evaluate(parse_results, context):
         return parse_results
 
     # recursion
+    if len(parse_results) != 3:
+        arg = parse_results[0]
+        condition_len = len(parse_results)
+        parse_list = []
+
+        # After the loop ends, a final evaluation is done
+        for i in range(int(condition_len/2-1)):
+            parse_list = [arg]
+            parse_list.extends(parse_results[2*i+1:2*i+3])
```

Comment:
Since you use `i` always with a factor of `2` maybe using `step=2` in the `range()` function would be better.